cmd/util/app.go

```diff
@@ -751,6 +753,8 @@ func ConstructSource(source *argoappv1.ApplicationSource, appOpts AppOptions, fl
 			setPluginOptEnvs(source, appOpts.pluginEnvs)
 		case "ref":
 			source.Ref = appOpts.ref
+		case "source-name":
+			source.Name = appOpts.SourceName
```

Comment:
We should be checking if the name is already assigned to some other source within the `app.spec.sources` before allowing to set it.

Reply:
I added the check in `validateAndNormalizeApp` to have a central validation for the both the UI and CLI API calls.